### ISSUE TITLE
can we get an element collector:  grand parent / great grand parent

### ISSUE THREAD

**Issue by royroque** [watir/watir#424]
can we get an element collector:  grand parent / great grand parent

I'm working on code when there is no unique name/id of the element; so as a workaround I end up using parent multiple times.

i.e.1
instead of
browser.label(:text=>'Username').parent.parent.text_field.set 'James'
USE:
browser.label(:text=>'Username').grandparent.text_field.set 'James'
browser.label(:text=>'Username').gparent.text_field.set 'James'
browser.label(:text=>'Username').parents[1].text_field.set 'James'

i.e.2
instead of
browser.label(:text=>'Username').parent.parent.parent.text_field.set 'James'
USE:
browser.label(:text=>'Username').greatgrandparent.text_field.set 'James'
browser.label(:text=>'Username').ggparent.text_field.set 'James'
browser.label(:text=>'Username').parents[2].text_field.set 'James'

**Comment by titusfortner** [comment]
Unfortunately have 2 different notions of `parent` in our code right now:
```
table = browser.table(id: 'foo')
td_element = table.td(id: 'bar')

td_element.parent.class == TableRow
td_element.instance_variable_get("@parent").class == Table
```

So, to start with, we should agree how we want to name things here.
Should `td_element.grandparent.class == Browser` or  `td_element.grandparent.class == Table` ?

I think `@parent` probably needs to get renamed to `@scope` or `@parent_scope`

The array format doesn't seem easy/desirable from an implementation standpoint, but I've been thinking about how we should switch over from the atom to use xpath for `#parent`. We could do something like this:
```
def ancestors(val)
  e = element(xpath: "./ancestor::*[#{val}]")
  e.exists? ? e.to_subtype : nil
end

def parent
  ancestors(1)
end

def grandparent
  ancestors(2)
end
```
I guess we could `#method_missing` the various `great`'s and `gg`'s as well...

**Comment by p0deje** [comment]
To me using lots of `parent` when locating elements is a code smell meaning that HTML should be fixed to be more testable. However, we still allow chaining parent calls, so I don't see a reason to have a bunch of new methods for this.

👎

**Comment by titusfortner** [comment]
I don't think we should assume that the tester of a website has control over the html under test.

**Comment by p0deje** [comment]
Okay, but I still believe that we should not complicate API with grand parents. This can easily be extended using separate gem or locally. I also don't see a problem with chaining parent calls.

**Comment by p0deje** [comment]
@titusfortner What's up with this?

**Comment by titusfortner** [comment]
This would be fixed by #451, which needs yardoc that I haven't gotten to, yet.

**Comment by titusfortner** [comment]
This was merged.
